File: chor-extraction/src/main/java/ast/cc/nodes/Condition.java
package ast.cc.nodes;

import ast.cc.CCVisitor;
import ast.cc.interfaces.CCNode;
import ast.cc.interfaces.Choreography;
import ast.sp.nodes.expr.BooleanExpression;
import epp.MergingException;

/**
 * Created by lara on 03/04/17.
 */
public class Condition implements Choreography{
    private final String process;
    private final BooleanExpression expression;
    private final CCNode thenChoreography;
    private final CCNode elseChoreograpy;

    public Condition(String process, BooleanExpression expression, CCNode thenChoreography, CCNode elseChoreograpy) {
        this.process = process;
        this.expression = expression;
        this.thenChoreography = thenChoreography;
        this.elseChoreograpy = elseChoreograpy;
    }

    public String getProcess() {
        return process;
    }

    public BooleanExpression getExpression() {
        return expression;
    }

    public CCNode getThenChoreography() {
        return thenChoreography;
    }

    public CCNode getElseChoreograpy() {
        return elseChoreograpy;
    }


    @Override
    public <T> T accept(CCVisitor<T> visitor) throws MergingException {
        return visitor.visit(this);
    }

    public String toString(){
        return "if " + process + "." + expression + " then " + thenChoreography.toString() + " else " + elseChoreograpy.toString();
    }
}

File: chor-extraction/src/main/java/ast/cc/nodes/Condition.kt
package ast.cc.nodes

import ast.cc.CCVisitor
import ast.cc.interfaces.CCNode
import ast.cc.interfaces.Choreography
import epp.MergingException

data class Condition(val process: String, val expression: String, val thenChoreography: CCNode, val elseChoreograpy: CCNode) : Choreography {


    @Throws(MergingException::class)
    override fun <T> accept(visitor: CCVisitor<T>): T {
        return visitor.visit(this)
    }

    override fun toString(): String {
        return "if " + process + "." + expression + " then " + thenChoreography.toString() + " else " + elseChoreograpy.toString()
    }
}
